src/applications/vaos/new-appointment/components/ClinicChoicePage.jsx

```diff
@@ -52,30 +52,32 @@ const uiSchema = {
   },
 };
 const pageKey = 'clinicChoice';
-export function ClinicChoicePage({
-  schema,
-  data,
-  facilityDetails,
-  typeOfCare,
-  clinics,
-  eligibility,
-  canMakeRequests,
-  openClinicPage,
-  updateFormData,
-  facilityDetailsStatus,
-  pageChangeInProgress,
-  routeToNextAppointmentPage,
-  routeToPreviousAppointmentPage,
-}) {
+export default function ClinicChoicePage() {
+  const {
+    data,
+    canMakeRequests,
+    clinics,
+    eligibility,
+    facilityDetails,
+    facilityDetailsStatus,
+    pageChangeInProgress,
+    schema,
+    typeOfCare,
+  } = useSelector(getClinicPageInfo(pageKey), shallowEqual);
+  // we may be  able to useCallback if the children useMemo to avoid re-renders
+  const dispatch = useDispatch();
   const history = useHistory();
   const typeOfCareLabel = formatTypeOfCare(typeOfCare.name);
   const usingUnsupportedRequestFlow =
     data.clinicId === 'NONE' && !canMakeRequests;
   const schemaAndFacilityReady =
     schema && facilityDetailsStatus !== FETCH_STATUS.loading;
-  useEffect(() => {
-    openClinicPage(pageKey, uiSchema, initialSchema);
-  }, []);
+  useEffect(
+    () => {
+      dispatch(actions.openClinicPage(pageKey, uiSchema, initialSchema));
+    },
+    [dispatch], // do we need to add dispatch as a dependency?
```

Comment:
I've been leaving out dependencies on these `useEffect`s that are meant to run once when the component mounts, since I think the intent is clearer. But it shouldn't break either way.

Reply:
Will do!